workspace/busStop.ts

```diff
@@ -0,0 +1,31 @@
+const request = require('request')
+import {Bus} from './bus'
+import * as _ from "lodash"
+
+export class BusStop {
+    constructor(public id: string, public name: string) {
+    }
+    getBuses(): Promise <Bus[]> {
+        let promiseLatAndLong: Promise <string> = new Promise((resolve) => {
+        var requestString = "https://api.tfl.gov.uk/StopPoint/" + 
+            this.id + "/Arrivals?app_id=ef0cca24&app_key=281fde5472478ee61c2d8a63c5e4f554"
+            request(requestString, function (error, response, body) {
+                resolve(body); 
+            });
+        });
+
+        return promiseLatAndLong.then((bodyString: string) => {
```

Comment:
I think everything on this return could be a separate method!

However, I'm really impressed with this. You've converted the JSON to an array of known objects _really early_, which is exactly what you should be doing with Typescripts (I'd possibly convert it even earlier and have a "LatLong" object, but I'm not going to make you do that).

Basically, high five.